src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/RuntimeLoaderPass.php

```diff
@@ -28,7 +28,17 @@ public function process(ContainerBuilder $container)
         $definition = $container->getDefinition('twig.runtime_loader');
         $mapping = array();
         foreach ($container->findTaggedServiceIds('twig.runtime') as $id => $attributes) {
-            $mapping[$container->getDefinition($id)->getClass()] = $id;
+            $def = $container->getDefinition($id);
+
+            if (!$def->isPublic()) {
+                throw new InvalidArgumentException(sprintf('The service "%s" must be public as it can be lazy-loaded.', $id));
+            }
+
+            if ($def->isAbstract()) {
+                throw new InvalidArgumentException(sprintf('The service "%s" must not be abstract as it can be lazy-loaded.', $id));
```

Comment:
The message here is not very accurate. An abstract service is not "get() able". It's not about lazy-loading.